tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +55,103 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new ApiWebResponse((HttpWebResponse)exception.Response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            _request.Method = "POST";
+            _request.ContentType = "multipart/form-data; boundary=" + Boundary;
+
+            using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                // Write form request using the boundary
+                var boundaryBytes = _boundarySeparatorInBytes ??= Encoding.ASCII.GetBytes(BoundarySeparator);
+                var trailerBytes = _boundaryTrailerInBytes ??= Encoding.ASCII.GetBytes(BoundaryTrailer);
+
+                // Write each MultipartFormItem
+                foreach (var item in items)
+                {
+                    byte[] headerBytes = null;
+
+                    // Check name is not null (required)
+                    if (item.Name is null)
+                    {
+                        Log.Warning("Error encoding multipart form item name is null. Ignoring item");
+                        continue;
+                    }
+
+                    // Ignore the item if the name contains ' or "
+                    if (item.Name.Contains("\"") || item.Name.Contains("'"))
+                    {
+                        Log.Warning("Error encoding multipart form item name: {Name}. Ignoring item.", item.Name);
+                        continue;
+                    }
+
+                    // Ignore the item if the string value cannot be encoded as ASCII
+                    if (Encoding.UTF8.GetByteCount(item.Name) != item.Name.Length)
```

Comment:
`Encoding.UTF8.GetByteCount()` allocates, is it worth doing here? Necessary I guess to ensure ascii only

Reply:
Well, if we want to be extra careful yes, but in the other hand, all these methods are internal, and we should know what we can do or not when implementing a new feature. What's your opinion on leaving a comment as a big WARNING about only supporting ASCII and remove the allocation?

Reply:
Given they're internal, I'd probably say go with the warning (in the doc-comments)? Arguably you could rid of all the validation that way if you want, but I'm inclined to keep the others as they're simple?

Reply:
done in [2268d49](https://github.com/DataDog/dd-trace-dotnet/pull/2962/commits/2268d494533e4ba931204b91aa131d204d1cf33e)